### PR TITLE
Add a jenkins proxy

### PR DESCRIPTION
**Pull request by 0xmichalis**
/cc @stevekuznetsov @csrwng

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -31,6 +31,11 @@ horologium:
 	oc process -f cluster/ci/config/prow/openshift/horologium.yaml | oc apply -f -
 .PHONY: horologium
 
+jenkins-proxy:
```
Comment (stevekuznetsov) [inline Makefile]: secrets?

File: tools/jenkins-proxy/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,7 @@
+FROM alpine:3.6
```
Comment (stevekuznetsov) [inline tools/jenkins-proxy/Dockerfile]: any reason not to use centos or the fedora minimal if you care about size?

Comment (stevekuznetsov) [inline tools/jenkins-proxy/Dockerfile]: any reason not to use centos or the fedora minimal if you care about size?

File: tools/jenkins-proxy/proxy.go [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+	if jobIndex+1 > len(parts) {
+		http.Error(w, "Forbidden.", http.StatusForbidden)
+		return
+	}
+
+	requestedJob := parts[jobIndex]
+	// Get the destination URL by looking at the job cache.
+	destURL := p.getDestURL(requestedJob)
+	if len(destURL) == 0 {
+		// Update the cache by relisting from all masters.
```
Comment (stevekuznetsov) [inline tools/jenkins-proxy/proxy.go]: what are we gaining from this cache if we relist on every access?

File: Makefile (hunk 1 above)
Comment (kargakis) [reply Makefile]: Yeah, missing secrets and config. Will add in a new push.

File: tools/jenkins-proxy/Dockerfile (hunk 2 above)
Comment (kargakis) [reply tools/jenkins-proxy/Dockerfile]: We don't really care about the size, I can use centos.

File: tools/jenkins-proxy/proxy.go (hunk 3 above)
Comment (kargakis) [reply tools/jenkins-proxy/proxy.go]: Newly added jobs in Jenkins can be picked up on the fly w/o the need to restart the proxy.

Comment (kargakis) [reply tools/jenkins-proxy/proxy.go]: *We are relisting on cache misses

Comment (stevekuznetsov) [reply tools/jenkins-proxy/proxy.go]: Oh, right, sorry misread that

Comment (stevekuznetsov) [reply tools/jenkins-proxy/proxy.go]: Oh, right, sorry misread that

**Comment by kargakis** [comment]
Comments addressed, more follow-up work to come.

**Pull request merged**